### ISSUE TITLE
[pbpaste on WSL2] I couldn't use pbpaste on wsl...

### ISSUE THREAD

**Issue by theAfricanQuant** [danielmiessler/fabric#269]
...But i found a workaround that could help

`win32yank.exe -o` accesses the clipboard of windows to extract teh info stored there

complete coda sample that worked for me: `win32yank.exe -o | fabric --pattern extract_main_idea`

**Comment by foxbg** [comment]
If you are using powershell Get-Clipboard is also good replacement. 
Source: https://labs.hbcom.info/posts/fabric/#additional-tips

**Comment by theNetworkChuck** [comment]
Using xsel covers it. Put this into your bashrc config

alias pbcopy='xsel --clipboard --input'
alias pbpaste='xsel --clipboard --output'

**Comment by theAfricanQuant** [comment]
I had tried the two above on wsl2...didn't work for me maybe I didn't understand how to use them

**Comment by theNetworkChuck** [comment]
Hello,

My name is Mikayla, I am NetworkChucks admin. Thanks for reaching out, for
any questions we do want to encourage you to use our discord as there are a
lot of people who would love to help you there! Here is the link:
https://discord.gg/networkchuck

Hope this helps,
Mikayla

**Comment by xhorntail** [comment]
> I had tried the two above on wsl2...didn't work for me maybe I didn't understand how to use them

The top two responses will only work outside of WSL.
Go with the @theNetworkChuck solution.

**Comment by omoralesg** [comment]
I am able to copy and paste my files within the command line in powershell or access the clipboard but when I run the paste with the | fabric -pattern custom_1 it says pattern not found .   Any ideas what am I doing wrong ? Thanks

**Comment by foxbg** [comment]
Is your pattern included when you run with --list
If custom look at https://github.com/danielmiessler/fabric?tab=readme-ov-file#custom-patterns

**Comment by omoralesg** [comment]
> Is your pattern included when you run with --list If custom look at https://github.com/danielmiessler/fabric?tab=readme-ov-file#custom-patterns

no, its not included in the --list .  When I put the command Get-Clipboard | fabric --pattern custom1   seems to run it but then I get the message, pattern not found

**Comment by foxbg** [comment]
> > Is your pattern included when you run with --list If custom look at https://github.com/danielmiessler/fabric?tab=readme-ov-file#custom-patterns
> 
> no, its not included in the --list . When I put the command Get-Clipboard | fabric --pattern custom1 seems to run it but then I get the message, pattern not found

Then it looks you want to use a custom pattern. In that case you need to review information in the link. (Open new issue as the problem you face does not seem to be related to the current one)

**Comment by ksylvan** [comment]
I am not sure if there is any question in here that still needs addressing. I'm closing this, but feel free to open an Issue or a Discussion.
